iroh/src/commands/sync.rs

```diff
@@ -226,25 +262,96 @@ impl Doc {
                 };
                 let mut stream = doc.get(filter).await?;
                 while let Some(entry) = stream.try_next().await? {
-                    println!("{}", fmt_entry(&entry));
+                    println!("{}", fmt_entry(entry.entry()));
+                }
+            }
+            Self::Watch { doc_id } => {
+                let doc = iroh.get_doc(env.doc(doc_id)?).await?;
+                let mut stream = doc.subscribe().await?;
+                while let Some(event) = stream.next().await {
+                    let event = event?;
+                    match event {
+                        iroh::sync::LiveEvent::InsertLocal { entry } => {
+                            println!("local change:  {}", fmt_entry(&entry))
+                        }
+                        iroh::sync::LiveEvent::InsertRemote {
+                            entry,
+                            from,
+                            content_status,
+                        } => {
+                            println!(
+                                "remote change: {} (via @{}, content {:?})",
+                                fmt_entry(&entry),
+                                fmt_short(from.as_bytes()),
+                                content_status
+                            )
+                        }
+                        iroh::sync::LiveEvent::ContentReady { hash } => {
+                            println!("content ready: {}", fmt_short(hash.as_bytes()))
+                        }
+                    }
+                }
+            }
+        }
+        Ok(())
+    }
+}
+
+#[derive(Debug, Clone, Parser)]
+pub enum AuthorCommands {
+    /// Set the active author (only works within the Iroh console).
+    Switch { id: AuthorId },
+    /// Create a new author.
+    Create {
+        /// Switch to the created author (only in the Iroh console).
+        #[clap(long)]
+        switch: bool,
+    },
+    /// List authors.
+    #[clap(alias = "ls")]
+    List,
+}
+
+impl AuthorCommands {
+    pub async fn run(self, iroh: &Iroh, env: ConsoleEnv) -> Result<()> {
+        match self {
+            Self::Switch { id } => {
+                env.set_author(id)?;
+            }
+            Self::List => {
+                let mut stream = iroh.list_authors().await?;
+                while let Some(author_id) = stream.try_next().await? {
+                    println!("{}", author_id);
```

Comment:
could we print out an indication if we have a secret key for the author? maybe sth like

```
@<author_id>   write ✔️ 
@<author_id>   write ✖️ 
```

Reply:
It only lists authors for which we have a secret key. `iroh_sync::Store::create_author/list_authors` only ever cares about full author keypairs. do we exepct this to change?
to list non-write authors, we'd have to iterate over all entries of all documents, wouldn't we?

Reply:
you are write, lets just document that a little clearer I guess 😅